### PR TITLE
[CWS] Add new process.is_thread field

### PR DESCRIPTION
**Pull request by YoannGh**
This PR adds the `process.is_thread` field to test whether a process is a child process that hasn't been replaced by another program.

Functional tests can be run with:
`inv -e security-agent.functional-tests --testflags "-test.v -test.run TestProcessIsThread"`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: docs/cloud-workload-security/agent_expressions.md [hunk 1]
```diff
@@ -262,6 +262,7 @@ The *file.rights* attribute can now be used in addition to *file.mode*. *file.mo
 | `process.fsuser` | string | FileSystem-user of the process |  |
 | `process.gid` | int | GID of the process |  |
 | `process.group` | string | Group of the process |  |
+| `process.is_thread` | bool | Indicator of whether the process is considered a thread (i.e. a child process that hasn't executed another program) |  |
```
Comment (urseberry) [inline docs/cloud-workload-security/agent_expressions.md:265]:
```suggestion
| `process.is_thread` | bool | Indicator of whether the process is considered a thread (that is, a child process that hasn't executed another program) |  |
```

File: docs/cloud-workload-security/secl.json [hunk 2]
```diff
@@ -583,6 +583,12 @@
           "definition": "Group of the process",
           "constants": ""
         },
+        {
+          "name": "process.is_thread",
+          "type": "bool",
+          "definition": "Indicator of whether the process is considered a thread (i.e. a child process that hasn't executed another program)",
```
Comment (urseberry) [inline docs/cloud-workload-security/secl.json:589]:
```suggestion
          "definition": "Indicator of whether the process is considered a thread (that is, a child process that hasn't executed another program)",
```

File: releasenotes/notes/cws-process-is-thread-5278ba17bd224e14.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    The CWS agent now supports filtering events depending on whether they are performed by a thread or not.
```
Comment (urseberry) [inline releasenotes/notes/cws-process-is-thread-5278ba17bd224e14.yaml:11]:
```suggestion
    The CWS agent now supports filtering events depending on whether they are performed by a thread.
```

**Comment by YoannGh** [comment]
Thanks for the review @urseberry, I updated the code documentation and generated the documentation files.

**Review by urseberry — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**